tests/Api/Product/ListProductApiIntegration.php

```diff
@@ -0,0 +1,1398 @@
+<?php
+
+namespace Akeneo\Pim\tests\Api\Product;
+
+use Akeneo\Pim\Pagination\PageInterface;
+use Akeneo\Pim\Pagination\ResourceCursorInterface;
+
+class ListProductApiIntegration extends AbstractProductApiTestCase
+{
+    public function testListPerPage()
+    {
+        $api = $this->createClient()->getProductApi();
+        $expectedProducts = $this->getExpectedProducts();
+        $baseUri = $this->getConfiguration()['api']['baseUri'];
+
+        $firstPage = $api->listPerPage(5);
+        $this->assertInstanceOf(PageInterface::class, $firstPage);
+        $this->assertNull($firstPage->getCount());
+        $this->assertNull($firstPage->getPreviousLink());
+        $this->assertNull($firstPage->getPreviousPage());
+        $this->assertFalse($firstPage->hasPreviousPage());
+        $this->assertTrue($firstPage->hasNextPage());
+        $this->assertSame($baseUri . '/api/rest/v1/products?page=2&with_count=false&pagination_type=page&limit=5', $firstPage->getNextLink());
+
+        $products = $firstPage->getItems();
+        $this->assertCount(5 ,$products);
+        for ($i = 0; $i < 5; $i++) {
```

Comment:
I would put it in a protected function `assertListResponse`.
```
$products = $firstPage->getItems();
$expectedProducts = array_slice($this->getExpectedProducts(), 0, 5);
$this->asssertListResponse($expectedProducts, $products);
```

And you can put the sanitize in the function.
It avoids `assertCount` as well.